### ISSUE TITLE
json error during kustomizationResourceDiff

### ISSUE THREAD

**Issue by wpbeckwith** [kbst/terraform-provider-kustomization#219]
We have a terraform config like

```
# Other modules and terraform
... 
module "foo" {
  helm_version = "1.2.0"
...
}

module "bar" {
  # this module uses the kbst provider
  depends_on [
    module.foo.release_metdata
  ]
}
```

The bar module, which installs fluxcd.io, does not use the any output of the foo module, it just has a dependency on 1 output to ensure that the foo module completes 1st.  Now when the terraform config is 1st created all is well.  And all subsequent plans are fine too, until I update the version # in the foo module to say 1.2.1.  Then the terraform plan will always fails with errors like 
```
╷
│ Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input
│ 
│   with module.cpu_dev2_eks.module.eks_cluster_init.module.fluxcd.kustomization_resource.p0["_/Namespace/_/flux-system"],
│   on .terraform/modules/cpu_dev2_eks.eks_cluster_init.fluxcd/fluxcd/main.tf line 59, in resource "kustomization_resource" "p0":
│   59: resource "kustomization_resource" "p0" {
│ 
╵
╷
│ Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input
│ 
│   with module.cpu_dev2_eks.module.eks_cluster_init.module.fluxcd.kustomization_resource.p0["apiextensions.k8s.io/CustomResourceDefinition/_/providers.notification.toolkit.fluxcd.io"],
│   on .terraform/modules/cpu_dev2_eks.eks_cluster_init.fluxcd/fluxcd/main.tf line 59, in resource "kustomization_resource" "p0":
│   59: resource "kustomization_resource" "p0" {
│ 
╵
╷
│ Error: github.com/kbst/terraform-provider-kustomize/kustomize.kustomizationResourceDiff: github.com/kbst/terraform-provider-kustomize/kustomize.(*kManifest).load: json error: unexpected end of JSON input
│ 
│   with module.cpu_dev2_eks.module.eks_cluster_init.module.fluxcd.kustomization_resource.p0["apiextensions.k8s.io/CustomResourceDefinition/_/alerts.notification.toolkit.fluxcd.io"],
│   on .terraform/modules/cpu_dev2_eks.eks_cluster_init.fluxcd/fluxcd/main.tf line 59, in resource "kustomization_resource" "p0":
│   59: resource "kustomization_resource" "p0" {
│ 
╵
...
```

This continues until I set the version in the foo module back to the original 1.2.0 and then the plans will successfully complete.

**Comment by pst** [comment]
I need a config to reproduce the issue to look into it.

**Comment by wpbeckwith** [comment]
Sure, I'll see if I can slim down our current config to a reproducible case.

**Comment by nazarewk** [comment]
Also getting blocked by this error, I've narrowed it down to empty `dm` (new value for manifest) string in here:
https://github.com/kbst/terraform-provider-kustomization/blob/1957b1eccaadadec5709675648a1860d67f6e56b/kustomize/resource_kustomization.go#L185-L191

**Comment by nazarewk** [comment]
not sure if it's way to go, have no time to dig into the root cause, but code changes in my PR fixed the issue without breaking anything (getting plan as expected)

**Comment by wpbeckwith** [comment]
I haven't narrowed this done to a smaller case ATM, been OOO for bereavement.  However, I did discover by accident that when I tried to run  a terraform plan several days later it worked.  Makes me think there is some type of caching being done and after a while the cache expires and a full set of data for the diff is sent.

**Comment by pst** [comment]
@nazarewk what we would need to figure out is if modified being empty is an expected state in Terraform or if there is some error causing this. If there is an error that causes modified to be empty, either in the manifests themselves or somewhere else along the way, just returning nil on the diff seems like a dangerous approach.

**Comment by nazarewk** [comment]
I am quite sure it is caused by ` -> (known after apply)`, but need to somehow confirm it

**Comment by nazarewk** [comment]
actually with my "fix" it did not update the values until the next run, so while it might be caused by `(known after apply)` there seems to be more to it

**Comment by nazarewk** [comment]
I've also noticed that current version does not notice manual changes done to resources showing no diff at all.

**Comment by pst** [comment]
> I've also noticed that current version does not notice manual changes done to resources showing no diff at all.

If the change did not update the annotation yes, that's a known limitation of the provider.

**Comment by CraigStuart** [comment]
I am running into this same issue. Is there a workaround I can look at?

**Comment by nazarewk** [comment]
not really, I used my locally built patch to work around it https://github.com/kbst/terraform-provider-kustomization/pull/221

**Comment by pascal-hofmann** [comment]
A workaround is to do targeted applies to resources that the broken resource depends upon. 
We frequently run into this issue and fixing this manually all the time is tedious. :(

**Comment by pascal-hofmann** [comment]
I created a repo for reproducing this issue: https://github.com/pascal-hofmann/terraform-provider-kustomization-reproduction-219

Other terraform providers treat this case like this:
```
	if !d.NewValueKnown("manifest") {
		// value is not known yet, so we can't diff anything.
		return nil
	}
```

I'm not sure how to live with the missing validation in this case, but I also wonder if `CustomizeDiff` is the right place for this in the first place.

**Comment by fatmcgav** [comment]
I've just hit the same issue... having to delete the managed resources or do a targetted apply is pretty cumbersome, especially so when the plan/apply is happening in a CI pipeline...
